File: Samples/Images/TiffToPdf/C#/TiffToPdf.cs
using System;
using System.Diagnostics;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class TiffToPdf
    {
        public static void Main()
        {
            // NOTE:
            // Without a license, the library won't allow you to create or read PDF documents.
            // To get a free time-limited license key, use the form on
            // https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE");

            var pathToFile = "TiffToPdf.pdf";

            using (var pdf = new PdfDocument())
            {
                string[] tiffFiles =
                {
                    @"..\Sample Data\multipage.tif",
                    @"..\Sample Data\pinkMask.tif"
                };

                int imagesAdded = 0;
                foreach (string tiff in tiffFiles)
                {
                    // open potentially multipage TIFF
                    PdfImageFrames frames = pdf.OpenImage(tiff);
                    foreach (PdfImageFrame frame in frames)
                    {
                        if (imagesAdded != 0)
                            pdf.AddPage();

                        PdfImage image = pdf.CreateImage(frame);
                        PdfPage pdfPage = pdf.Pages[pdf.PageCount - 1];

                        // adjust page size, so image will occupy the whole page
                        pdfPage.Height = image.Height;
                        pdfPage.Width = image.Width;
                        pdfPage.Canvas.DrawImage(image, 0, 0, pdfPage.Width, pdfPage.Height, 0);

                        imagesAdded++;
                    }
                }

                pdf.Save(pathToFile);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(pathToFile) { UseShellExecute = true });
        }
    }
}

File: Samples/Images/TiffToPdf/VB.NET/TiffToPdf.vb
Imports BitMiracle.Docotic
Imports BitMiracle.Docotic.Pdf

Namespace BitMiracle.Docotic.Pdf.Samples
    Public Class TiffToPdf
        Public Shared Sub Main()
            ' NOTE:
            ' Without a license, the library won't allow you to create or read PDF documents.
            ' To get a free time-limited license key, use the form on
            ' https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE")

            Dim pathToFile As String = "TiffToPdf.pdf"

            Using pdf As New PdfDocument
                Dim tiffFiles() As String = New String() {
                    "..\Sample Data\multipage.tif",
                    "..\Sample Data\pinkMask.tif"
                }
                Dim imagesAdded As Integer = 0
                For Each tiff As String In tiffFiles
                    ' open potentially multipage TIFF
                    Dim frames As PdfImageFrames = pdf.OpenImage(tiff)
                    If frames Is Nothing Then
                        Console.WriteLine("Cannot add image")
                        Return
                    End If

                    For Each frame As PdfImageFrame In frames
                        If imagesAdded <> 0 Then
                            pdf.AddPage()
                        End If

                        Dim image As PdfImage = pdf.AddImage(frame)
                        Dim pdfPage As PdfPage = pdf.Pages(pdf.PageCount - 1)
                        ' adjust page size, so image fill occupy whole page
                        pdfPage.Height = image.Height
                        pdfPage.Width = image.Width
                        pdfPage.Canvas.DrawImage(image, 0, 0, pdfPage.Width, pdfPage.Height, 0)
                        imagesAdded += 1
                    Next
                Next
                pdf.Save(pathToFile)
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")

            Process.Start(New ProcessStartInfo(pathToFile) With {.UseShellExecute = True})
        End Sub
    End Class
End Namespace